Makefile

```diff
@@ -381,6 +381,7 @@ partialclean::
 	rm -f compilerlibs/ocamlopttoplevel.cmxa
 
 ocamlnat: compilerlibs/ocamlcommon.cmxa compilerlibs/ocamloptcomp.cmxa \
+    compilerlibs/ocamlbytecomp.cmxa \
```

Comment:
The second commit of this PR is an experiment: `Compdynlink` is used instead of `Dynlink` in `toplevel/*.ml`, so removing the dependency on `dynlink.cmxa` for `ocamlnat`.

Does that make sense ? I tested it and it works, but I couldn't test it on Windows (I am on holidays).